projects/ngx-favicon/src/lib/ngx-favicon.service.ts

```diff
@@ -1,6 +1,51 @@
-import { Injectable } from '@angular/core';
+import { Injectable, Renderer2, ElementRef, Inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { NGX_FAVICON_CONFIG } from './ngx-favicon.tokens';
+import { NgxFaviconConfig, Dictionnary } from './ngx-favicon.interface';
 
+class NgxFaviconUnknownFaviconError extends Error {
+  constructor(faviconName: string) {
+    super(
+      `Favicon with name "${faviconName}" was not found into the "custom" property of the config`,
+    );
+  }
+}
+
+// https://github.com/angular/angular/issues/20351#issuecomment-344009887
+/** @dynamic */
 @Injectable()
-export class NgxFaviconService {
-  constructor() {}
+export class NgxFaviconService<CustomFavicon extends string> {
+  constructor(
+    @Inject(DOCUMENT) private document: Document,
+    @Inject(NGX_FAVICON_CONFIG)
+    private ngxFaviconConfig: NgxFaviconConfig<Dictionnary<string>>,
+  ) {}
+
+  public setCustomFavicon(faviconName: CustomFavicon): void {
+    if (!faviconName) {
```

Comment:
```suggestion
    if (!faviconName || !this.ngxFaviconConfig.custom[faviconName]) {
```

Reply:
I've added a test for that case and made the change at the same time :)